### PR TITLE
chore: replace globby with tinyglobby

### PR DESCRIPTION
**Pull request by benmccann**
## PR Checklist

- [x] Addresses an existing open issue: fixes https://github.com/typescript-eslint/typescript-eslint/issues/9453
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Replaces `globby` with `tinyglobby` for a smaller more efficient implementation with fewer dependencies

### CODE REVIEW

File: packages/typescript-estree/tests/lib/parse.test.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ jest.mock('globby', () => {
   const globby = jest.requireActual<typeof globbyModule>('globby');
```
Comment (bradzacher) [inline packages/typescript-estree/tests/lib/parse.test.ts:43]: This would also need to change

**Review by bradzacher — CHANGES_REQUESTED** [review]

File: packages/typescript-estree/tests/lib/parse.test.ts (hunk 1 above)
Comment (benmccann) [reply packages/typescript-estree/tests/lib/parse.test.ts]: done. thanks!

**Comment by Josh-Cena** [comment]
Why do we have both this and https://github.com/typescript-eslint/typescript-eslint/pull/9518?

**Comment by benmccann** [comment]
Oh, thanks for pointing that out. I hadn't seen it. I think `tinyglobby` is better as it has many fewer dependencies, but that PR is passing the CI whereas this one isn't. I'll check if the author is able to switch to `tinyglobby` in the other PR and if so perhaps we can close this one

**Comment by SukkaW** [comment]
It seems that your CI failed due to incorrect spying of glob functions. Might wanna check that out.

**Comment by benmccann** [comment]
> I just want to point out that this change added 2 new dependencies to the project and did not actually remove any included dep, if you check yarn.lock. So this feels kinda pointless change this time.
(note, I’m just a user of this plugin, so my opinions do not reflect what the maintainers may think)

The `yarn.lock` isn't what's important here because it includes both dependencies and `devDependencies`. `globby` remains only for developers of this project. If you look at where it's used now it's in things like docusaurus. This PR would remove about 40% of dependencies for all consumers of `typescript-eslint` going from 44 down to 26

**Comment by benmccann** [comment]
> It seems that your CI failed due to incorrect spying of glob functions. Might wanna check that out.

I found the line I missed updating before. It's now correctly mocking. However, it's returning things in a different order. I think we basically have to combine our two PRs to get it working

**Comment by SukkaW** [comment]
> However, it's returning things in a different order.

See https://github.com/typescript-eslint/typescript-eslint/pull/9518#issuecomment-2308461397

**Comment by bradzacher** [comment]
As per https://github.com/typescript-eslint/typescript-eslint/issues/9453#issuecomment-2309080805 we're going to go with #9518 for now.
We may revisit this in future and switch from `fast-glob` to `tinyglobby`.

**Pull request closed without merging**